packages/meditrak-app/app/assessment/selectors.js

```diff
@@ -109,3 +110,85 @@ export const getQuestion = (state, questionId) => state.assessment.questions[que
 export const getAnswers = state => state.assessment.answers;
 
 export const getAnswerForQuestion = (state, questionId) => getAnswers(state)[questionId];
+
+export const getArithmeticResult = (state, arithmeticQuestionId) => {
+  const { config } = getQuestion(state, arithmeticQuestionId);
+
+  const {
+    arithmetic: { formula, valueTranslation = {}, defaultValues = {}, answerDisplayText = '' },
+  } = config;
+
+  const values = {};
+  const expressionParser = new ExpressionParser();
+  const variables = expressionParser.getVariables(formula);
+  let translatedAnswerDisplayText = answerDisplayText;
+
+  const getArithmeticAnswer = questionId => {
+    const answer = getAnswerForQuestion(state, questionId);
+
+    if (valueTranslation[questionId] && valueTranslation[questionId][answer] !== undefined) {
+      return valueTranslation[questionId][answer]; // return translated answer if there's any
+    }
+
+    if (answer !== undefined) {
+      return answer; // return raw answer
+    }
+
+    return defaultValues[questionId] !== undefined ? defaultValues[questionId] : 0; // No answer found, return the default answer
+  };
+
+  // Setting up scope values.
+  variables.forEach(questionIdVariable => {
+    const questionId = questionIdVariable.replace(/^\$/, ''); // Remove the first $ prefix to get the actual questionId
+    values[questionIdVariable] = getArithmeticAnswer(questionId);
+  });
+
+  // Replace variables with real values in answerDisplayText
+  variables.forEach(questionIdVariable => {
+    const questionId = questionIdVariable.replace(/^\$/, ''); // Remove the first $ prefix to get the actual questionId
+    const answer = getArithmeticAnswer(questionId);
+    translatedAnswerDisplayText = translatedAnswerDisplayText.replace(questionId, answer);
+  });
+
+  expressionParser.setScope(values);
+
+  const result = !isNaN(expressionParser.evaluate(formula))
+    ? Math.round(expressionParser.evaluate(formula) * 1000) / 1000 // Round to 3 decimal places
+    : 0;
+  translatedAnswerDisplayText = translatedAnswerDisplayText.replace('$result', result);
+
+  return {
+    answerDisplayText: translatedAnswerDisplayText,
+    result,
+  };
+};
+
+export const getConditionResult = (state, conditionQuestionId) => {
+  const { config } = getQuestion(state, conditionQuestionId);
+  const {
+    condition: { conditions },
+  } = config;
+  const expressionParser = new BooleanExpressionParser();
+  const checkConditionMet = ({ formula, defaultValues }) => {
+    const values = {};
+    const variables = expressionParser.getVariables(formula);
+
+    variables.forEach(questionIdVariable => {
+      const questionId = questionIdVariable.replace(/^\$/, ''); // Remove the first $ prefix
+      const answer = getAnswerForQuestion(state, questionId);
+      const value = answer || defaultValues[questionId];
```

Comment:
What if it had been answered "false" but the default value was "true"?

Oh, I guess we use strings rather than booleans, so even "false" will be "truthy"